spring-cloud-gcp-starters/spring-cloud-gcp-starter-config/src/main/resources/META-INF/spring.factories

```diff
@@ -0,0 +1,4 @@
+org.springframework.cloud.bootstrap.BootstrapConfiguration=\
```

Comment:
I'm still wrapping my head around the bootstrap context concept which I hadn't seen before. I guess we need this here because of that, but I think it would also be nice to add `org.springframework.boot.autoconfigure.EnableAutoConfiguration` to this file, so that the IDEs can auto-complete the name of the properties.

I tried this locally but the IDE still isn't auto-completing the property name. I'm still trying to figure out why...

Reply:
For the config client, there is no value in providing the `EnableAutoConfiguration`, since its done in the application context, post bootstrap
Are you talking of providing an empty org.springframework.boot.autoconfigure.EnableAutoConfiguration ?

Reply:
If I recall correctly, pointing `EnableAutoConfiguration` here to our auto configuration class makes IDEs auto-suggest property names when you're typing.

But when I'm editing `application.properties`, if I add `spring.cloud.gcp.c`, the editor doesn't auto-suggest the option (as it does for sql and core).

Reply:
I'm not sure why after adding the following to this file, the IDE still won't recognise the spring.cloud.gcp.config property prefix.

```
org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
  org.springframework.cloud.gcp.config.autoconfig.GcpConfigAutoConfiguration
```

@artembilan do you have any idea why?

Reply:
auto-configuration doesn't enable ide autocompetion of properties. Property metadata, does that.

Reply:
Says [here](https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html#configuration-metadata) that `@ConfigurationProperties` should take care of generating the configuration metadata, but my browser isn't reading it..

Reply:
https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html#configuration-metadata-annotation-processor

Reply:
+1 what Spencer says. 
In our language it sounds like: "I told you!" 

https://github.com/spring-cloud/spring-cloud-gcp/issues/47